protocol/contracts/beanstalk/field/FieldFacet.sol

```diff
@@ -167,89 +183,122 @@ contract FieldFacet is Invariable, ReentrancyGuard {
+        s.activeField = fieldId;
+        emit ActiveFieldSet(fieldId);
     }
 
     //////////////////// GETTERS ////////////////////
 
     /**
-     * @notice Returns the total number of Pods ever minted.
+     * @notice Returns the total number of Pods ever minted in the Field.
+     * @param fieldId The index of the Field to query.
      */
-    function podIndex() public view returns (uint256) {
-        return s.field.pods;
+    function podIndex(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].pods;
     }
 
     /**
      * @notice Returns the index below which Pods are Harvestable.
+     * @param fieldId The index of the Field to query.
      */
-    function harvestableIndex() public view returns (uint256) {
-        return s.field.harvestable;
+    function harvestableIndex(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].harvestable;
     }
 
     /**
      * @notice Returns the number of outstanding Pods. Includes Pods that are
      * currently Harvestable but have not yet been Harvested.
+     * @param fieldId The index of the Field to query.
      */
-    function totalPods() public view returns (uint256) {
-        return s.field.pods.sub(s.field.harvested);
+    function totalPods(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].pods - s.fields[fieldId].harvested;
     }
 
     /**
      * @notice Returns the number of Pods that have ever been Harvested.
+     * @param fieldId The index of the Field to query.
      */
-    function totalHarvested() public view returns (uint256) {
-        return s.field.harvested;
+    function totalHarvested(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].harvested;
     }
 
     /**
      * @notice Returns the number of Pods that are currently Harvestable but
      * have not yet been Harvested.
      * @dev This is the number of Pods that Beanstalk is prepared to pay back,
      * but that haven’t yet been claimed via the `harvest()` function.
+     * @param fieldId The index of the Field to query.
      */
-    function totalHarvestable() public view returns (uint256) {
-        return s.field.harvestable.sub(s.field.harvested);
+    function totalHarvestable(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].harvestable - s.fields[fieldId].harvested;
     }
 
     /**
      * @notice Returns the number of Pods that are not yet Harvestable. Also known as the Pod Line.
+     * @param fieldId The index of the Field to query.
      */
-    function totalUnharvestable() public view returns (uint256) {
-        return s.field.pods.sub(s.field.harvestable);
+    function totalUnharvestable(uint256 fieldId) public view returns (uint256) {
+        return s.fields[fieldId].pods - s.fields[fieldId].harvestable;
+    }
+
+    function isHarvesting(uint256 fieldId) public view returns (bool) {
```

Comment:
could we add natspec here?

Reply:
added
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d65b6bacc7e6d96105a48c759af2f5222b0ccb26#diff-d1df16bc4f6f3b8f37c60a5dab5717e751f8a2986236fc936965d2ecd0f0bdf6R301-R304